yb-voyager/src/dbzm/dbzm.go

```diff
@@ -64,12 +76,80 @@ func findDebeziumDistribution() error {
 	return nil
 }
 
+func GenerateAndStoreYBStreamID(config Config) (string, error) {
+	err := findDebeziumDistribution(config.SourceDBType)
+	if err != nil {
+		return "", fmt.Errorf("error in finding debezium distribution: %s", err)
+	}
+	YB_CLIENT_WRAPPER_JAR := fmt.Sprintf("%s/yb-client-cdc-stream-wrapper.jar", DEBEZIUM_DIST_DIR) //$DEBEZIUM_DIST/yb-client-cdc-stream-wrapper.jar
+	tableName := strings.Split(config.TableList[0], ".")[1]                                        //any table name in the database is required by yb-client createCDCStream(...) API
+	command := fmt.Sprintf("java -jar %s -master_addresses %s -table_name %s -db_name %s ", YB_CLIENT_WRAPPER_JAR, config.YBServers, tableName, config.DatabaseName)
+
+	if config.SSLRootCert != "" {
+		command += fmt.Sprintf(" -ssl_cert_file %s", config.SSLRootCert)
+	}
+
+	cmd := exec.CommandContext(context.Background(), "/bin/bash", "-c", command)
+	var outbuf bytes.Buffer
+	var errbuf bytes.Buffer
+	cmd.Stdout = &outbuf
+	cmd.Stderr = &errbuf
+	err = cmd.Start()
+	if err != nil {
+		if outbuf.String() != "" {
+			log.Infof("Output of the command %s: %s", command, outbuf.String())
+		}
+		log.Infof("Failed to start command: %s, error: %s", command, err)
+		return "", err
+	}
+	err = cmd.Wait()
+	if err != nil {
+		if outbuf.String() != "" {
+			log.Infof("Output of the command %s: %s", command, outbuf.String())
+		}
+		log.Infof("Failed to wait for command: %s , error: %s", command, err)
+		return "", err
+	}
+	//output of yb-admin command - CDC Stream ID: <stream_id>
+	rgx := regexp.MustCompile(`CDC Stream ID: ([a-zA-Z0-9]+)`)
+	matches := rgx.FindStringSubmatch(outbuf.String())
+	if len(matches) != 2 {
+		return "", fmt.Errorf("error in parsing output of command: %s, output: %s", command, outbuf.String())
+	}
+	streamID := matches[1]
+	//save streamID in a file
+	streamIDFile := filepath.Join(config.ExportDir, "metainfo", "yb_cdc_stream_id.txt")
+	file, err := os.Create(streamIDFile)
+	if err != nil {
+		return "", fmt.Errorf(" creating file: %s, error: %s", streamIDFile, err)
+	}
+	defer file.Close()
+	_, err = file.WriteString(streamID)
+	if err != nil {
+		return "", fmt.Errorf(" writing to file: %s, error: %s", streamIDFile, err)
+	}
+	return streamID, nil
+}
+
+func ReadYBStreamID(exportDir string) (string, error) {
+	streamIDFilePath := filepath.Join(exportDir, "metainfo", "yb_cdc_stream_id.txt")
+	if utils.FileOrFolderExists(streamIDFilePath) {
+		file, err := os.ReadFile(streamIDFilePath)
+		if err != nil {
+			return "", fmt.Errorf("failed to read stream id file: %w", err)
+		}
+		return string(file), nil
+	}
+	return "", fmt.Errorf("yugabytedb cdc stream id not found at %s. Please use --start-clean to start the streaming", streamIDFilePath)
```

Comment:
If stream-id is not found, the code should generate one. No need of passing `--start-clean`.

If `--start-clean` is passed and there is already existing stream-id, we should see if we can "delete" the stream-id from the server itself.

Reply:
added delete option for start-clean